packages/nx/src/adapter/ngcli-adapter.ts

```diff
@@ -51,10 +53,73 @@ import {
   CustomHasher,
   Executor,
   ExecutorConfig,
+  ExecutorContext,
   ExecutorsJson,
   TaskGraphExecutor,
 } from '../config/misc-interfaces';
 import { readPluginPackageJson } from '../utils/nx-plugin';
+import { WorkspaceNodeModulesArchitectHost as AngularWorkspaceNodeModulesArchitectHost } from '@angular-devkit/architect/node';
+
+export async function createBuilderContext(
+  builderInfo: {
+    builderName: string;
+    description: string;
+    optionSchema: any;
+  },
+  context: ExecutorContext
+) {
+  require('nx/src/adapter/compat');
+  const fsHost = new NxScopedHost(context.root);
+  const { workspace } = await workspaces.readWorkspace(
+    'angular.json',
+    workspaces.createWorkspaceHost(fsHost)
+  );
+  const architectHost = new WrappedWorkspaceNodeModulesArchitectHost(
+    workspace,
+    context.root
+  );
+
+  const registry = new schema.CoreSchemaRegistry();
+  registry.addPostTransform(schema.transforms.addUndefinedDefaults);
+  registry.addSmartDefaultProvider('unparsed', () => {
+    // This happens when context.scheduleTarget is used to run a target using nx:run-commands
+    return [];
+  });
+  const { Architect } = require('@angular-devkit/architect');
+
+  const architect: Architect = new Architect(architectHost, registry);
+
+  const builderContext: import('@angular-devkit/architect').BuilderContext = {
+    workspaceRoot: context.root,
+    target: {
+      project: context.projectName,
+      target: context.targetName,
+      configuration: context.configurationName,
+    },
+    builder: {
+      ...builderInfo,
+    },
+    logger: getLogger(),
+    id: 1,
+    currentDirectory: process.cwd(),
+    scheduleTarget: architect.scheduleTarget,
+    addTeardown(teardown: () => Promise<void> | void) {},
+    validateOptions: architect['_scheduler'].validateOptions,
+    getBuilderNameForTarget: architectHost.getBuilderNameForTarget,
+    scheduleBuilder: architect.scheduleBuilder,
+    getTargetOptions: architectHost.getOptionsForTarget,
+    reportProgress: architect['_scheduler'].reportProgress,
+    reportRunning: architect['_scheduler'].reportRunning,
```

Comment:
Similar to the previous comment, the scheduler does not define these two functions. Unlike the previous comment, we could provide noop functions for them because we don't use them (similar to `reportStatus` below).